pkg/cloudprovider/providers/rackspace/rackspace.go

```diff
@@ -391,3 +466,146 @@ func (os *Rackspace) GetZone() (cloudprovider.Zone, error) {
 
 	return cloudprovider.Zone{Region: os.region}, nil
 }
+
+// Create a volume of given size (in GiB)
+func (os *Rackspace) CreateVolume(name string, size int, tags *map[string]string) (volumeName string, err error) {
+	return "", errors.New("unimplemented")
+}
+
+func (os *Rackspace) DeleteVolume(volumeName string) error {
+	return errors.New("unimplemented")
+}
+
+// Attaches given cinder volume to the compute running kubelet
+func (os *Rackspace) AttachDisk(diskName string) (string, error) {
+	instanceID, err := readInstanceID()
+	if err != nil {
+		return "", err
+	}
+
+	disk, err := os.getVolume(diskName)
+	if err != nil {
+		return "", err
+	}
+
+	compute, err := os.getComputeClient()
+	if err != nil {
+		return "", err
+	}
+
+	if len(disk.Attachments) > 0 && disk.Attachments[0]["server_id"] != nil {
+		if instanceID == disk.Attachments[0]["server_id"] {
+			glog.V(4).Infof("Disk: %q is already attached to compute: %q", diskName, instanceID)
+			return disk.ID, nil
+		} else {
+			errMsg := fmt.Sprintf("Disk %q is attached to a different compute: %q, should be detached before proceeding", diskName, disk.Attachments[0]["server_id"])
+			glog.Errorf(errMsg)
+			return "", errors.New(errMsg)
+		}
+	}
+
+	_, err = volumeattach.Create(compute, instanceID, &volumeattach.CreateOpts{
```

Comment:
should this really be `github.com/rackspace/gophercloud/openstack/compute/v2/extensions/volumeattach`? I would rely on the `rackspace` namespace here, since you're not handling openstack types